stetho/src/main/java/com/facebook/stetho/common/android/AccessibilityUtil.java

```diff
@@ -0,0 +1,252 @@
+import android.view.ViewParent;
+import android.widget.AdapterView;
+import android.widget.HorizontalScrollView;
+import android.widget.ScrollView;
+import android.widget.Spinner;
+
+import java.util.List;
+
+/**
+ * This class provides utility methods for determining certain accessibility properties of
+ * {@link View}s and {@link AccessibilityNodeInfoCompat}s. It is porting some of the checks from
+ * {@link com.googlecode.eyesfree.utils.AccessibilityNodeInfoUtils}, but has stripped many features
+ * which are unnecessary here.
+ */
+public final class AccessibilityUtil {
+  private AccessibilityUtil() {
+  }
+
+  /**
+   * Returns whether the specified node has text or a content description.
+   *
+   * @param node The node to check.
+   * @return {@code true} if the node has text.
+   */
+  public static boolean hasText(AccessibilityNodeInfoCompat node) {
+    if (node == null) {
+      return false;
+    }
+
+    return !TextUtils.isEmpty(node.getText()) || !TextUtils.isEmpty(node.getContentDescription());
+  }
+
+  /**
+   * Returns whether the supplied {@link View} and {@link AccessibilityNodeInfoCompat} would
+   * produce spoken feedback if it were accessibility focused.  NOTE: not all speaking nodes are
+   * focusable.
+   *
+   * @param view The {@link View} to evaluate
+   * @param node The {@link AccessibilityNodeInfoCompat} to evaluate
+   * @return {@code true} if it meets the criterion for producing spoken feedback
+   */
+  public static boolean isSpeakingNode(AccessibilityNodeInfoCompat node, View view) {
+    if (node == null || view == null) {
+      return false;
+    }
+
+    if (!node.isVisibleToUser()) {
+      return false;
+    }
+
+    int important = ViewCompat.getImportantForAccessibility(view);
+    if (important == ViewCompat.IMPORTANT_FOR_ACCESSIBILITY_NO_HIDE_DESCENDANTS ||
+        (important == ViewCompat.IMPORTANT_FOR_ACCESSIBILITY_NO &&
+            node.getChildCount() <= 0)) {
+      return false;
+    }
+
+    return hasText(node) || node.isCheckable() || hasNonActionableSpeakingDescendants(node, view);
+  }
+
+  /**
+   * Determines if the supplied {@link View} and {@link AccessibilityNodeInfoCompat} has any
+   * children which are not independently accessibility focusable and also have a spoken
+   * description.
+   * <p>
+   * NOTE: Accessibility services will include these children's descriptions in the closest
+   * focusable ancestor.
+   *
+   * @param view The {@link View} to evaluate
+   * @param node The {@link AccessibilityNodeInfoCompat} to evaluate
+   * @return {@code true} if it has any non-actionable speaking descendants within its subtree
+   */
+  public static boolean hasNonActionableSpeakingDescendants(
+      AccessibilityNodeInfoCompat node, View view) {
+
+    if (node == null || view == null || !(view instanceof ViewGroup)) {
+      return false;
+    }
+
+    ViewGroup viewGroup = (ViewGroup) view;
+    for (int i = 0; i < viewGroup.getChildCount(); i++) {
```

Comment:
for the loop, prefer this slightly more efficient form:

`for (int i = 0, count = viewGroup.getChildCount(); i < count; i++) {`

Then we don't have a virtual function call for every loop iteration!